boards: arm: stm32u5a9j-dk: fix missing pull down on switch

gzzi opened the pull request:
Hello,

The user switch on the stm32u5a9j_dk board require a pull down on mcu side. Otherwise a short push will last longer than expected. This is mentioned in [ST um2967-discovery-kit-with-stm32u5a9nj-mcu-stmicroelectronics.pdf](https://www.st.com/resource/en/user_manual/um2967-discovery-kit-with-stm32u5a9nj-mcu-stmicroelectronics.pdf) at 12.11:

> The user button schematics have been optimized, so it is up to the user to manage the internal pull-down and
debounce filtering by software

gzzi committed:
boards: arm: stm32u5a9j-dk: fix switch gpio config

The user switch on the board require a pull down on mcu side and
is active high.

Signed-off-by: Guillaume Galeazzi <guillaume.galeazzi@gmail.com>

ajarmouni-st commented:
@gzzi in section "12.11 User button function", it says "The user button (B1) is a way to interact with software. This is a selection key, active high, which is mapped on a WKUP PIO (PC13)." so shouldn't we put GPIO_ACTIVE_HIGH instead of LOW, w/t PULL flag?

gzzi commented:
@ajarmouni-st yes I think it's also more correct to set it as ACTIVE_HIGH. With this change, `gpio_pin_get_dt` of the switch return 0 when none is pressing the button. I changed commit message and added your suggestion.

ajarmouni-st commented:
Tested, can be merged.
